projects/packages/sync/src/class-dedicated-sender.php

```diff
@@ -58,9 +58,9 @@ public static function spawn_sync( $queue ) {
 			return new WP_Error( 'empty_queue_' . $queue->id );
 		}
 
-		// Return early if we've gotten a retry-after header response.
+		// Return early if we've gotten a retry-after header response that is not expired.
 		$retry_time = get_option( Actions::RETRY_AFTER_PREFIX . $queue->id );
-		if ( $retry_time ) {
+		if ( $retry_time && $retry_time >= microtime( true ) ) {
```

Comment:
@bisko This should be enough, right? Wondering whether we should return error and update the option instead as we already do in `do_sync_and_set_delays` [here](https://github.com/Automattic/jetpack/blob/master/projects/packages/sync/src/class-sender.php#L384)

Reply:
I'm not a big fan of the logic as it is right now. This should be part of the queue locking and not being checked/updated separately in multiple places.

This fix should suffice for now and we can rework the checks for locks at a later stage, when we reach an agreement on the right approach.